### ISSUE TITLE
Serialization does not store hashobj, and deserialization always results in using sha1

### ISSUE THREAD

**Issue by ekzhu** [ekzhu/datasketch#18]
There is a bug in the serialization / deserialization - hashobj is not stored, so it will always revert back to sha1 (the __init__ default argument) when the class is deserialized.

**Comment by ekzhu** [comment]
Currently the user can get around this by explicitly setting the hashobj member after deserialization. For example if you use [pyhash](https://pypi.python.org/pypi/pyhash/0.8):

```python
import pyhash
# Assume `data` is serialized MinHash
minhash = pickle.loads(data)
minhash.hashobj = pyhash.murmur3_32()
```

**Comment by ekzhu** [comment]
This bug is fixed in version 1.1.1. But now the old serialized MinHash won't be properly deserialized in the newer versions. Migration can be done as follows:

1. Use `pip install datasketch=1.1.0` to switch to version 1.1.0, which still uses the old deserialization method.
2. Deserialize your `MinHash` objects using `pickle` (if you used the now-deprecated `serialize` method for serialization, you need to use `MinHash.deserialize`), and then convert them into [`LeanMinHash`](https://ekzhu.github.io/datasketch/documentation.html#lean-minhash).
```python
from datasketch import *
# ...
# Assume minhash is a MinHash object you have deserialized
lean = LeanMinHash(minhash)
```
3. Use `pickle.dumps` or the `serialize` method to serialize the `LeanMinHash` objects to where you saved your old `MinHash` objects.
4. Use `pip install datasketch -U` to upgrade to the latest version.

[`LeanMinHash`](https://ekzhu.github.io/datasketch/documentation.html#lean-minhash) supports all methods of `MinHash` except for `update`. So if you don't need `update` this should be all you need to do. If update is required, you can convert it back to 
`MinHash` object (see [here](https://ekzhu.github.io/datasketch/documentation.html#lean-minhash)).
